### DISCUSSION TITLE
Minio Docker with TLS

### DISCUSSION THREAD

**Discussion by Twilek-de** [minio/minio#11395] [Q&A] [answered]
Hi all,
I try to setup Minio via Docker and secure it with TLS. There is no official documented way to do this. I have tried to map the /root/.minio directory and put some certs there but it doesn´t seem to have any effect. Is there an official way to this?

**Comment by Twilek-de** [comment]
I have helped myself by using Apache as an reverse Proxy, but it still seems worthwhile to make the inbuild TLS funcitonality usable in Docker.

**Comment by nitisht** [comment, marked as answer]
Setting up with Docker is explained here https://github.com/minio/minio/issues/6820 

As long as you follow https://docs.min.io/docs/how-to-secure-access-to-minio-server-with-tls.html and put certs in correct locations you should be fine.

**Comment by Twilek-de** [comment]
What I did is mapping the /root/.minio/ directory outside of the container. It then generated the certs directory there. Then I put my certs in there as explained in the guide and spun up a new container. But that does not work, you only get http access, not https.

**Reply by iHarshalAgarwal** [reply]
Hi, did you get this working ? I am trying to the same but got no success.
I mounted the /root/.minio directory to the host and placed cert created using https://docs.min.io/docs/how-to-secure-access-to-minio-server-with-tls.html#generate-private-key-with-rsa

**Reply by Twilek-de** [reply]
Nope I used Apache as a proxy....

**Comment by panpan0000** [comment]
I finally got it. 
the doc and help says   `--certs-dir value, -S value  path to certs directory (default: "/root/.minio/certs")`
But even I put my certs in that folder: 
```
sh-4.4# ls /root/.minio/certs/CAs/
  private.key	public.crt
```
You will have to explicitly specify  `--certs-dir` to your CA path.

Below are my docker run command just for your reference:

```
docker run --name minio -p 443:443 \
 -p 9000:9000   -p 9001:9001  \
--env MINIO_ACCESS_KEY=xxx --env MINIO_SECRET_KEY=xxxx  \
  -v /root/.minio/certs/:/root/.minio/certs/  \
 minio/minio server /data  \
--console-address ":9001"  \
--address ":443"  \
--certs-dir /root/.minio/certs/CAs/
```
